### PR TITLE
Important fixes to NOT handling in xlucene

### PR DESCRIPTION
**Pull request by peterdemartini**
@jsnoble and I fixed an issue with handling chained NOT expressions. Additionally, I added another option to the luncene query access library for allowing implicit querie and I found and fixed problems with nested queries in the translator.

### CODE REVIEW

File: packages/xlucene-evaluator/src/document-matcher/index.ts [hunk 1]
```diff
@@ -36,75 +38,66 @@ export default class DocumentMatcher {
...
-            let addParens = false;
-            let negation  = isNegation || false;
-            let negateLeftExp = false;
             let fnStr = '';
 
-            if (node.operator === 'NOT') {
-                negation = true;
-            }
+            logger.trace('build fn for node', {
+                field,
```
Comment (jsnoble) [inline packages/xlucene-evaluator/src/document-matcher/index.ts]: this trace needs to be removed

File: packages/xlucene-evaluator/src/document-matcher/index.ts [hunk 2]
```diff
@@ -118,21 +111,23 @@ export default class DocumentMatcher {
         argsFns.push(_.get);
         strFnArgs.push('data', `return ${fnStr}`);
 
+        logger.trace('created filter fn', fnStr);
```
Comment (jsnoble) [inline packages/xlucene-evaluator/src/document-matcher/index.ts]: should remove trace

File: packages/xlucene-evaluator/src/lucene-query-parser/index.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export default class LuceneQueryParser {
         try {
             this._ast = parser.parse(luceneStr);
             logger.trace(`parsed ${luceneStr} to `, this._ast);
+            // logger.trace(`parsed ${luceneStr} to `, JSON.stringify(this._ast, null, 4));
```
Comment (jsnoble) [inline packages/xlucene-evaluator/src/lucene-query-parser/index.ts]: remove trace and commented out code

File: packages/xlucene-evaluator/src/document-matcher/index.ts (hunk 2 above)
Comment (peterdemartini) [reply packages/xlucene-evaluator/src/document-matcher/index.ts]: I believe this log message is useful information for debugging document match queries and requires `env DEBUG='*document-matcher* DEBUG_LOG_LEVEL='trace'`.

**Pull request merged**